src/hooks/useActionCenterSubscriber/useTradeActionSubscriber.ts

```diff
@@ -0,0 +1,212 @@
+        metadata: {
+          lifiRoute: isLifiTradeQuote(tradeQuote) ? tradeQuote.selectedLifiRoute : undefined,
+          chainflipSwapId: firstStep?.chainflipSpecific?.chainflipSwapId,
+          stepIndex: currentHopIndex,
+        },
+        status: SwapStatus.Pending,
+      }
+
+      dispatch(swapSlice.actions.upsertSwap(swap))
+
+      dispatch(
+        actionCenterSlice.actions.upsertAction({
+          type: ActionCenterType.Swap,
+          status: ActionStatus.Pending,
+          title: translate('notificationCenter.notificationTitle', {
+            sellAmountAndSymbol: toCrypto(
+              fromBaseUnit(
+                firstStep.sellAmountIncludingProtocolFeesCryptoBaseUnit,
+                tradeSellAsset.precision,
+              ),
+              tradeSellAsset.symbol,
+              {
+                maximumFractionDigits: 8,
+                omitDecimalTrailingZeros: true,
+                abbreviated: true,
+                truncateLargeNumbers: true,
+              },
+            ),
+            buyAmountAndSymbol: toCrypto(
+              fromBaseUnit(lastStep.buyAmountAfterFeesCryptoBaseUnit, tradeBuyAsset.precision),
+              tradeBuyAsset.symbol,
+              {
+                maximumFractionDigits: 8,
+                omitDecimalTrailingZeros: true,
+                abbreviated: true,
+                truncateLargeNumbers: true,
+              },
+            ),
+          }),
+          metadata: {
+            swapId: swap.id,
+          },
+          assetIds: [tradeSellAsset.assetId, tradeBuyAsset.assetId],
+          initiatorAccountId: sellAccountId,
+        }),
+      )
+    }
+  }, [
+    tradeExecution,
+    dispatch,
+    translate,
+    toCrypto,
+    tradeSellAsset,
+    tradeBuyAsset,
+    tradeQuote,
+    previousTradeExecutionState,
+    previousTradeQuoteId,
+    currentHopIndex,
+    previousTxHash,
+    sellAccountId,
+    previousQuoteOrRate,
+  ])
+
+  // Update swap with tx hash or swapId for chainflip when the user did the first step
+  useEffect(() => {
+    if (!tradeExecution) return
+    if (!tradeExecution.firstHop.swap.sellTxHash) return
+    if (!tradeQuote) return
+
+    const swap = selectSwapByQuoteId(store.getState(), {
+      quoteId: tradeQuote.id as unknown as QuoteId,
+    })
+
+    if (
+      tradeQuote.quoteOrRate === 'quote' &&
+      tradeExecution.state === TradeExecutionState.FirstHop &&
+      tradeExecution.firstHop.swap.sellTxHash &&
+      swap?.status === SwapStatus.Pending
+    ) {
+      dispatch(
+        swapSlice.actions.updateSwap({
+          id: swap.id,
+          sellTxHash: tradeExecution.firstHop.swap.sellTxHash,
+        }),
+      )
+    }
+  }, [tradeExecution, dispatch, tradeQuote, previousTxHash])
+
+  // Update actions status when swap is confirmed or failed
+  const actionsQueries = useMemo(() => {
+    return pendingSwapActions
+      .map(action => {
+        switch (action.type) {
+          case ActionCenterType.Swap: {
```

Comment:
Looks like we shouldn't need it since those are already swap actions?
Should we improve narrowing?